### PR TITLE
Flatpak workshop project. Some additional questions.

### PR DESCRIPTION
**Pull request by panossot**
During a workshop for flatpak, I tried to flatpack a fork of https://github.com/temandr/Sudoku-puzzle (https://github.com/panossot/Sudoku-puzzle). 

First of all the original project has no licence, so maybe it cannot be uploaded to flathub. 

I have some additional questions though:
Now, it works, because I copy the java .class files in the sandbox. But the idea is to run also the build.sh file in order to generate them each time (https://github.com/panossot/Sudoku-puzzle/blob/master/build.sh#L3).  I was thinking that line : "append-path": "/usr/lib/sdk/openjdk9/bin" at the build options of the .json manifest would do the trick, but javac is not available in the sandbox. How can this be resolved?

Thank you in advance,
Panagiotis

### CODE REVIEW

File: Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  ],
+  "finish-args": [
+    "--share=network",
+    "--share=ipc",
+    "--socket=x11",
+    "--socket=wayland",
+    "--filesystem=xdg-run/dconf",
+    "--filesystem=~/.config/dconf:ro",
+    "--talk-name=ca.desrt.dconf",
+    "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
```
Comment (nedrichards) [inline Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: You're going to want to add:
>     "--env=PATH=/usr/bin:/app/bin:/usr/lib/sdk/openjdk9/bin"
here

Comment (panossot) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: still I get "./build.sh: line 3: javac: command not found"
after adding also "./build.sh" in 
"name": "sudoku-puzzle",
      "buildsystem": "simple",
      "build-commands": [ "./build.sh",
	"rm -rf ./bin/META-INF"
      ],

Comment (TingPing) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: finish args are for run time not build time.

Comment (panossot) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: I think that the pb is that javac is not included in the sandbox

Comment (Alexander-Wilms) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: Maybe this fixes it: https://github.com/flathub/org.octave.Octave/blob/master/org.octave.Octave.json#L27

Comment (panossot) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: Now I get ...

Running: /usr/lib/sdk/openjdk9/install.sh
/usr/lib/sdk/openjdk9/install.sh: line 2: mkdir: command not found
/usr/lib/sdk/openjdk9/install.sh: line 4: cp: command not found
/usr/lib/sdk/openjdk9/install.sh: line 5: rm: command not found
/usr/lib/sdk/openjdk9/install.sh: line 6: cp: command not found
Error: module openjdk: Child process exited with code 127

Comment (Alexander-Wilms) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: Looks like append-path replaces the existing PATH variable, which removes /usr/bin

Comment (panossot) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: It seems that this lines that I added removed the /usr/bin :
"env": {
      "PATH": "/usr/bin:/app/bin:/usr/lib/sdk/gfortran-62/bin:/usr/lib/sdk/openjdk9/bin"
    }
the "append-path": "/usr/lib/sdk/openjdk9/bin", should do the same, shouldn't? So, I removed "PATH": "/usr/bin:/app/bin:/usr/lib/sdk/gfortran-62/bin:/usr/lib/sdk/openjdk9/bin" and now I get again the "./build.sh: line 3: javac: command not found"

Comment (panossot) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: The pb should be the the extension is sdk. 

 "org.freedesktop.Sdk.Extension.openjdk9"

Isn't there a jdk extension?

Comment (nedrichards) [reply Sudoku/Sudoku-Puzzle/org.gnome.Sudoku-Puzzle.json]: Filed an issue: https://github.com/flathub/org.freedesktop.Sdk.Extension.openjdk9/issues/3

**Pull request closed without merging**